packages/text/src/TextStyle.ts

```diff
@@ -773,7 +847,7 @@ function getColor(color)
  * @param {Array} array2 Second array to compare
  * @return {boolean} Do the arrays contain the same values in the same order
  */
-function areArraysEqual(array1, array2)
+function areArraysEqual(array1: any[], array2: any[]): boolean
```

Comment:
I think it should be generic function

Reply:
Changing this to `function areArraysEqual<T1, T2>(array1: T1[], array2: T2[]): boolean` results in a problem with the !== comparison: `This condition will always return 'true' since the types 'T1' and 'T2' have no overlap.`

I can change it to `function areArraysEqual<T>(array1: T[], array2: T[]): boolean` but then it cannot be called with different types of arrays. Then again maybe you don't need to because if they're different types then they definitely are not equal. I'll go with this one.

Reply:
Why need to compare arrays with different type?